docs/examples/junit4/generic/src/test/java/generic/support/TestSpecificImageNameSubstitutor.java

```diff
@@ -0,0 +1,34 @@
+package generic.support;
+
+import org.testcontainers.utility.DefaultImageNameSubstitutor;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+/**
+ * An {@link ImageNameSubstitutor} which makes it possible to use fake image names in
+ * {@link generic.ImageNameSubstitutionTest}. This implementation simply reverses a fake image name when presented, and
+ * is hardcoded to act upon the specific fake name in that test.
+ */
+public class TestSpecificImageNameSubstitutor extends ImageNameSubstitutor {
+
+    private final DefaultImageNameSubstitutor defaultImageNameSubstitutor = new DefaultImageNameSubstitutor();
+
+    @Override
+    public DockerImageName apply(final DockerImageName original) {
+        if (original.equals(DockerImageName.parse("registry.mycompany.com/mirror/mysql:8.0.22"))) {
+            return defaultImageNameSubstitutor.apply(DockerImageName.parse("mysql"));
+        } else {
+            return defaultImageNameSubstitutor.apply(original);
```

Comment:
why not just `return original`? Given the priority of `1`, I thought that `TestSpecificImageNameSubstitutor` will go after the default (`0`) one :)

Reply:
As we've updated to not use service loader or priorities, this no longer applies 😄